### PR TITLE
seeded_rand_string() function

### PR DESCRIPTION
**Pull request by pegasd**
This function is useful for generating consistent random strings (read "passwords") across different nodes.

### CODE REVIEW

File: lib/puppet/functions/seeded_rand_string.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+#   seeded_rand_string(5, '', 'abcdef')
+Puppet::Functions.create_function(:seeded_rand_string) do
+  # @param length Length of string to be generated.
+  # @param seed Seed string.
+  # @param charset String that contains characters to use for the random string.
+  #
+  # @return [String] Random string.
+  dispatch :rand_string do
+    param 'Integer[1]', :length
+    param 'String', :seed
```
Comment (hlindberg) [inline lib/puppet/functions/seeded_rand_string.rb]: The text description states that seed is optional, but this declaration is for a required param.

**Comment by hlindberg** [comment]
I think it is worth opening a ticket in puppet's Jira in the MODULES project to discuss the merit of this function. There are already other functions related to 'random' and there may be opportunity to do something about sprawling additions of "random this" and "random that"...

**Comment by eputnam** [comment]
@hlindberg we already have an fqdn_rand() and fqdn_rand_string() but only seeded_rand() with no seeded_rand_string(). This method doesn't seem like sprawl with that in mind.

**Comment by hlindberg** [comment]
`fqdn_rand_string` can be given a `seed` - how is this different? Why are so many different functions needed for almost the same functionality? Why not a simpler API and more intuitive names? Say `random_number` and `random_string` and then use named options to make those produce the different variations of using fqdn as seed or some other seed, length of string, an Integer or a Float etc. Come to think of it, even just `random` would work and then give it a data type that expresses a range - for example `random(Integer[1,100])`, `random(String[10])`.

**Comment by pegasd** [comment]
@hlindberg There is currently no way to generate the same randomized string on two different hosts.
I do agree with your naming approach though. Let's pick whichever one is more suitable for this.

**Comment by david22swan** [comment]
@pegasd Is there any movement on this PR?

**Comment by pegasd** [comment]
@david22swan How should I proceed here? I'm open to suggestions. Or, if the community finds this insecure, I'm OK with closing this for good.

**Comment by hlindberg** [comment]
Hm, I had objections to this, those concerns were not addressed. We could do much better than what was now merged. Suggest reverting this.

**Pull request merged**